packages/api/src/modules/providers/sirene/stockUniteLegale/sireneStockUniteLegale.service.ts

```diff
@@ -1,88 +1,34 @@
-import * as fs from "fs";
-import StreamZip from "node-stream-zip";
 import sireneUniteLegaleDbPort from "../../../../dataProviders/db/sirene/stockUniteLegale/sireneStockUniteLegale.port";
-import sireneStockUniteLegaleApiPort from "../../../../dataProviders/api/sirene/sireneStockUniteLegale.port";
+import uniteLegalEntreprisesService from "../../uniteLegalEntreprises/uniteLegal.entreprises.service";
+import { UniteLegalEntrepriseEntity } from "../../../../entities/UniteLegalEntrepriseEntity";
+import uniteLegalNameService from "../../uniteLegalName/uniteLegal.name.service";
+import { SireneStockUniteLegaleEntity } from "../../../../entities/SireneStockUniteLegaleEntity";
 import { SireneUniteLegaleDbo } from "./@types/SireneUniteLegaleDbo";
 import SireneStockUniteLegaleParser from "./parser/sireneStockUniteLegale.parser";
+import SireneStockUniteLegaleAdapter from "./adapter/sireneStockUniteLegale.adapter";
 
 export class SireneStockUniteLegaleService {
-    private directory_path;
-
-    private getOrCreateDirectory() {
-        if (!fs.existsSync(this.directory_path)) {
-            this.directory_path = fs.mkdtempSync(__dirname + "/tmpSirene");
-        }
-    }
-
-    public async getAndParse() {
-        await this.getExtractAndSaveFiles();
-        console.log("start getExtractAndSaveFiles");
-        await SireneStockUniteLegaleParser.parseCsvAndInsert(this.directory_path + "/StockUniteLegale_utf8.csv");
-        this.deleteTemporaryFolder();
+    public async parse(filePath: string) {
+        await SireneStockUniteLegaleParser.parseCsvAndInsert(
+            filePath,
+            this._saveBatchAssoData.bind(this),
```

Comment:
pourquoi on a besoin de ce `bind(this)` ?

Reply:
on appel déjà une méthode de la classe le `this` représentera la classe, non ?

Reply:
en plus, pour le call à `saveBatchNonAssoData`, il n'y a même pas de référence à `this`

Reply:
Ça plante sans 🤷 j'avoue j'ai pas investigué plus à partir du moment où j'ai trouvé une solution

Reply:
plus qu'étrange ! je vais investiguer car souvent quand on comprend pas un truc il peut y avoir anguille sous rouche.. ^^'

Reply:
du coup après check je comprend le `bind.this` pour le premier avec la ref à `this.insertMany`.

Ce que je comprend pas encore c'est pourquoi on a laissé passer ce choix archi avec un parser qui reçoit un callback d'insertion en bdd.. on pouvait pas retourner le contenu parsé (trop gros ?) ou nous même renvoyé un stream au service pour que ce soit plus propre ? on utilise rarement ce genre de manip

Reply:
On peut clairement pas retourner le contenu parsé parce que trop gros. C'est moi qui introduit ce choix d'archi. J'ai envisagé de retourner un stream mais soit ça laisse au service le soin de fermer et gérer les erreurs du stream qui est ouvert par ailleurs (et donc pas propre), soit le parseur fait tout et j'aimais pas non plus

Reply:
Je vois bien les problèmes, c'est juste que c'est plus vraiment un parser du coup ! ;)

Reply:
Je crée un petit ticket si un jour j'ai envie d'y réfléchir :p